### PR TITLE
[Improvement][dao,server] unit test for ConditionsTask

### PR DESCRIPTION
**Pull request by hsupu**
## What is the purpose of the pull request

refactor the unit test for ConditionsTask

## Brief change log

- refactor and enable the unit test for ConditionsTask
- add FAILURE dependence test for ConditionsTask
- fix a potential NPE at TaskInstance.getDependence and add a unit test for it
- enable unit test for TaskInstance

## Verify this pull request

This change added tests and can be verified as follows:

- Updated `org.apache.dolphinscheduler.server.master.ConditionsTaskTest`
- Updated `org.apache.dolphinscheduler.dao.entity.TaskInstanceTest`
- Tested locally

### CODE REVIEW

File: dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/entity/TaskInstance.java [hunk 1]
```diff
@@ -16,19 +16,18 @@
...
 import org.apache.dolphinscheduler.common.model.TaskNode;
-import org.apache.dolphinscheduler.common.utils.*;
+import org.apache.dolphinscheduler.common.utils.JSONUtils;
+
 import com.baomidou.mybatisplus.annotation.IdType;
+import com.baomidou.mybatisplus.annotation.TableField;
 import com.baomidou.mybatisplus.annotation.TableId;
 import com.baomidou.mybatisplus.annotation.TableName;
-import org.apache.dolphinscheduler.common.utils.JSONUtils;
+import com.fasterxml.jackson.annotation.JsonFormat;
```
Comment (Jave-Chen) [inline dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/entity/TaskInstance.java:31]: Why change common.utils.JSONUtils to jackson.annotation.JsonFormat ?

Comment (hsupu) [reply dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/entity/TaskInstance.java:31]: A import refactoring happened here.

**Comment by Jave-Chen** [comment]
Sonar Analysis failed, you can see more at
https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=3385&metric=new_coverage&view=list

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/Constants.java [hunk 2]
```diff
@@ -391,7 +391,12 @@ private Constants() {
     public static final int RPC_PORT = 50051;
 
     /**
-     * forbid running task
+     * task can be run normally
+     */
+    public static final String FLOWNODE_RUN_FLAG_NORMAL = "NORMAL";
```
Comment (yangyichao-mango) [inline dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/Constants.java:396]: Hi,
Good job, but there is no necessary to add the `FLOWNODE_RUN_FLAG_NORMAL` in Constants just for test case.
If you want to add the constant, you can directly add `FLOWNODE_RUN_FLAG_NORMAL` in `ConditionsTaskTest`

File: dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/entity/TaskInstance.java [hunk 3]
```diff
@@ -16,19 +16,18 @@
...
```
Comment (yangyichao-mango) [inline dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/entity/TaskInstance.java:30]: The `JsonFormat` is not unused. Please delete.

File: dolphinscheduler-server/src/test/java/org/apache/dolphinscheduler/server/master/ConditionsTaskTest.java [hunk 4]
```diff
@@ -17,7 +17,16 @@
 package org.apache.dolphinscheduler.server.master;
```
Comment (yangyichao-mango) [inline dolphinscheduler-server/src/test/java/org/apache/dolphinscheduler/server/master/ConditionsTaskTest.java:17]: Hi,
I checkout to your branch and found that the `ConditionsTaskTest.java` is not in the test case includer.
Please add the test to root pom.
image

**Comment by yangyichao-mango** [comment]
Hi,
Please add test case for your new code in `TaskInstance` to resolve this sonar fail.
image

**Comment by hsupu** [comment]
> Please add test case for your new code in `TaskInstance` to resolve this sonar fail.

done.

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/Constants.java (hunk 2 above)
Comment (hsupu) [reply dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/Constants.java]: I just moved it to the test file by now, but I think it should be a constant for `"NORMAL"` in `common/Constants.java` for use in future.

**Comment by hsupu** [comment]
@yangqinlong There is a big change in `pom.xml` to sort tests, plz check.

File: pom.xml [hunk 5]
```diff
@@ -783,12 +783,38 @@
...
+                        <!--<include>**/dao/entity/ProcessDefinitionTest.java</include>-->
+                        <include>**/dao/entity/TaskInstanceTest.java</include>
+                        <include>**/dao/entity/UdfFuncTest.java</include>
                         <include>**/dao/mapper/AccessTokenMapperTest.java</include>
                         <include>**/dao/mapper/AlertGroupMapperTest.java</include>
                         <include>**/dao/mapper/CommandMapperTest.java</include>
                         <include>**/dao/mapper/ConnectionFactoryTest.java</include>
                         <include>**/dao/mapper/DataSourceMapperTest.java</include>
-                        <include>**/dao/entity/UdfFuncTest.java</include>
+                        <include>**/dao/mapper/DataSourceUserMapperTest.java</include>
```
Comment (yangyichao-mango) [inline pom.xml:797]: Please revert the changes and just add your test.

Comment (hsupu) [reply pom.xml]: done. and you may wannt to know that there're two parts for `/dao` here and the `dao/entity/ProcessDefinitionTest` is missing.
